tket/test/src/test_FrameRandomisation.cpp

```diff
@@ -358,5 +360,32 @@ SCENARIO(
   }
 }
 
+SCENARIO("Frame Randomisation Segmentation Fault") {
+  // https://github.com/CQCL/tket/issues/1015
+  PauliFrameRandomisation pfr;
+  Circuit c(4);
+  c.add_op<unsigned>(OpType::CX, {0, 2});
+  c.add_op<unsigned>(OpType::Z, {0});
+  c.add_op<unsigned>(OpType::CX, {0, 3});
+  c.add_op<unsigned>(OpType::Z, {0});
+  c.add_op<unsigned>(OpType::CX, {0, 3});
+  c.add_op<unsigned>(OpType::CX, {0, 1});
+  std::vector<Circuit> circs = pfr.sample_randomisation_circuits(c, 1);
+  CHECK(circs.size() == 1);
+}
+
+SCENARIO("Frame Randomisation non-real DAG") {
+  // https://github.com/CQCL/tket/issues/1015
+  PauliFrameRandomisation pfr;
+  Circuit c(4);
+  c.add_op<unsigned>(OpType::CnX, {0, 1, 2, 3});
+  CompilationUnit cu(c);
+  RebaseTket()->apply(cu);
+  std::vector<Circuit> circs =
+      pfr.sample_randomisation_circuits(cu.get_circ_ref(), 1);
+  CHECK(circs.size() == 1);
+  TKET_ASSERT(circs[0].get_commands().size() == 107);
```

Comment:
Shouldn't this be a `CHECK`, rather than ` `TKET_ASSERT`?

Reply:
Ah yes, sorry I went quite heavy with the `TKET_ASSERT` while debugging.

Reply:
Updated